### ISSUE TITLE
Add Black as our Python formatter 

### ISSUE THREAD

**Issue by C0D3D3V** [ddnet/ddnet#5481]
Would be nice if we could use a formatter for python files, agree on project settings for the formatter and also add it to `scripts/fix_style.py`.

There are two well known Python formatter: [autopep8](https://github.com/hhatto/autopep8) and [black](https://github.com/psf/black).

I would recommend `black` because I have good experiences with it, and I think black has the better default rules but both are good formatter, and we probably want to customize the formatting rules anyway.

**Comment by Chairn** [comment]
:+1: although i don't have any preference on the formatter.

**Comment by SollyBunny** [comment]
#5661 was rejected
I am in favour of autopep8 but anything is fine, we need a formatter

**Comment by heinrich5991** [comment]
> we need a formatter

Why do you want to have a formatter?

**Comment by SollyBunny** [comment]
> Why do you want to have a formatter?

The python we have doesn't follow anything
There's hundreds of (AI written) articles on why formatters are good
We also format our CPP so that's something

**Comment by heinrich5991** [comment]
I'm not interested in why others want a formatter, but you specifically (since you said "we need a formatter").

**Comment by heinrich5991** [comment]
(https://docs.astral.sh/ruff/ seems to allow more configuration, maybe we could fit it to our existing codebase to minimize the diff.)

**Comment by SollyBunny** [comment]
1. maintainability, multiple people working on parts near eachother or fast forwarding through history will have less conflicts

2. readability, going imbetween multiple styles is very jarring (seeing tabs and spaces inconsistnently used hurts my soul)

* ![Image](https://github.com/user-attachments/assets/b5dbea77-fad7-4e63-80a7-c3278a608bce)
* ![Image](https://github.com/user-attachments/assets/27c4a279-8f77-457f-a0f2-4fac73aa71e8)
* `'` vs `"`, we seem to use `'''` for multiline all of the time, `"` for strings sometimes

3. ease of writing, i dont have to think where spaces go i just format it and it's right

**Comment by heinrich5991** [comment]
Thanks!

**Comment by AssassinTee** [comment]
Everyone is using 4 indent style and all modern formatters enforce it (you can't set it, this is also true for ruff).
I'd prefer this over the current state of manually needing to do line formatting.

Whatever formatter I use, even with `yapf`, the diffs are huge. I suggest switching to a modern formatter like ruff, but doing this bit by bit for the directories instead of introducing one big diff

**Comment by heinrich5991** [comment]
Ruff seems to support configuring it: https://docs.astral.sh/ruff/settings/#format_indent-style.

**Comment by AssassinTee** [comment]
you are right, it does - we should introduce it with as minimal changes as possible and work from there.

I'd want the python files to follow the pep8 standard in the long run (even with 4 indents), as it's just more maintainable and we have more tooling available.

**Comment by heinrich5991** [comment]
`ruff` adds two spaces in front of each inline comment. This is not the current style, but it can't be configured in `ruff`: https://github.com/ddnet/ddnet/pull/11447#issuecomment-3664733902.
